packages/delegator-e2e/test/delegationManagement.test.ts

```diff
@@ -0,0 +1,450 @@
+import { beforeEach, test, expect } from 'vitest';
+import {
+  createExecution,
+  createDelegation,
+  Implementation,
+  toMetaMaskSmartAccount,
+  ExecutionMode,
+  type MetaMaskSmartAccount,
+} from '@metamask/delegation-toolkit';
+import {
+  createCaveatBuilder,
+  encodePermissionContexts,
+  encodeExecutionCalldatas,
+  getDelegationHashOffchain,
+} from '@metamask/delegation-toolkit/utils';
+import {
+  DeleGatorCore,
+  DelegationManager,
+} from '@metamask/delegation-toolkit/contracts';
+import {
+  transport,
+  gasPrice,
+  sponsoredBundlerClient,
+  deploySmartAccount,
+  deployCounter,
+  CounterContract,
+  publicClient,
+  fundAddress,
+} from './utils/helpers';
+import { expectUserOperationToSucceed } from './utils/assertions';
+import { createClient, encodeFunctionData, parseEther, type Hex } from 'viem';
+import { generatePrivateKey, privateKeyToAccount } from 'viem/accounts';
+import { chain } from '../src/config';
+import CounterMetadata from './utils/counter/metadata.json';
+
+let aliceSmartAccount: MetaMaskSmartAccount<Implementation.Hybrid>;
+let bobSmartAccount: MetaMaskSmartAccount<Implementation.Hybrid>;
+let aliceCounter: CounterContract;
+
+/**
+ * These tests verify the delegation management functionality:
+ * - disableDelegation: Allows a delegator to disable a delegation they've created
+ * - enableDelegation: Allows a delegator to re-enable a previously disabled delegation
+ * - disabledDelegations: Allows checking the status of delegations
+ *
+ * Test scenarios:
+ * 1. Alice creates a delegation to Bob
+ * 2. Bob uses the delegation successfully
+ * 3. Alice disables the delegation
+ * 4. Bob attempts to use the disabled delegation (should fail)
+ * 5. Alice re-enables the delegation
+ * 6. Bob uses the delegation again successfully
+ */
+
+beforeEach(async () => {
+  const client = createClient({ transport, chain });
```

Comment:
this should be `createPublicClient` - but you can also just `import { publicClient } from './helpers';`